plugins/destination/file/client/write.go

```diff
@@ -26,9 +34,45 @@ func (c *Client) WriteTableBatch(ctx context.Context, table *schema.Table, data
 	if err != nil {
 		return err
 	}
-	defer f.Close()
 
-	return c.Client.WriteTableBatchFile(f, table, data)
+	h, err := c.Client.WriteHeader(f, table)
+	if err != nil {
+		return err
+	}
+
+	key := mapKey(sourceSpec, table)
+	c.tableFiles[key] = writer{
+		f: f,
+		h: h,
+	}
+	c.lastSpec = sourceSpec
+
+	return nil
+}
+
+func (c *Client) CloseTable(_ context.Context, sourceSpec specs.Source, table *schema.Table, _ time.Time) error {
+	c.logger.Debug().Str("source", sourceSpec.Name).Str("table", table.Name).Msg("CloseTable")
+
+	c.tableFilesMu.Lock()
```

Comment:
can sync.Map be used instead (we can use https://pkg.go.dev/sync#Map.LoadAndDelete for this one)?